### DISCUSSION TITLE
Use This Package for Pivot Table

### DISCUSSION THREAD

**Discussion by annis-souames** [spatie/laravel-permission#1778] [Q&A] [answered]
I have many-to-many relationship (user-team), so a user can join multiple teams and a team can have multiple users, at first I was  adding permissions only in team model, but I need to add it in team-user pivot table through this package, so a user can have different permissions for each team he has joined ? Any Ideas how can I achieve this.

**Comment by bhushan** [comment]
facing same kind of issue ..

**Comment by brandon-9** [comment]
Crude, and early, but I have the same situation. I created a "teamAccess" model/table to store the roles.
id | team_id | user_id

I check the active team (team ID 1) stored on the User table, then going through the team, I find the teamAccess where team_id = user_active_team_id

User Model:
public function teamAccess()
    {
        return $this->hasMany(TeamAccess::class);
    }
    
    
   Then I have a Policy with a condition:
   $user->teamAccess()->where('team_id', $user->team_id)->first()->hasPermissionTo('create')
   
   
Might get you somewhere...

**Comment by sys-auditing** [comment]
Hi @annis-souames, did you figure out the solution ?

**Comment by Hesesses** [comment]
I'm working on something similar deciding between this package or creating something custom...

Could this be handled with subparts?
https://spatie.be/docs/laravel-permission/v4/basic-usage/wildcard-permissions#subparts

`$user->givePermissionTo('posts.*.1,4,6');` 

You would specify the wanted team_ids

**Comment by kaaaaaaaaaaai** [comment]
@annis-souames @bhushan @sys-auditing @Hesesses @brandon-9 
I'm working on something similar.
Did you find a solution?

**Comment by erikn69** [comment, marked as answer]
[discussions 1782](https://github.com/spatie/laravel-permission/discussions/1782)
That's my solution, it's not the best way to do it, but it's been working for me for years.

**Reply by AlexanderFalkenberg** [reply]
@erikn69 can you tell me what usua_id, empr_id in your screenshot  mean and what they are refencering to. I'm a little confused.
![image](https://user-images.githubusercontent.com/35661687/176905252-f9ddcea1-2d87-4def-8b29-1b84f87ddf92.png)

**Reply by erikn69** [reply]
Really old answer, i make this feature native 
https://github.com/spatie/laravel-permission/pull/1804
[teams-permissions](https://spatie.be/docs/laravel-permission/v5/basic-usage/teams-permissions)
